Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -0,0 +1,583 @@
+                    {
+                        var createdMatch = await _matchService.PostMatchAsync(new MatchUploadViewModel
+                        {
+                            StartTime = day,
+                            SeriesID = seriesID,
+                            SurfaceID = availableSurfaces[surfaceIndex].SurfaceID,
+                            TeamIDs = teamIDs
+                        });
+                        createdMatches.Add(createdMatch);
+                        surfaceIndex++;
+                    }
+                    i++;
+                    j--;
+                }
+                var temp = teams[0];
+                teams.Add(temp);
+                teams.RemoveAt(0);
+            }
+            return createdMatches;
+        }
+
+        //rudamentary implementation (only allows all teams into 1 match)
+        private async Task<IEnumerable<MatchViewModel>> ScheduleLadderAsync(int seriesID)
+        {
+            var createdMatches = new List<MatchViewModel>();
+            var teams = _context.SeriesTeam
+                .Where(st => st.ID == seriesID)
+                .Select(st => st.ID);
+            var series = _context.Series
+                .FirstOrDefault(s => s.ID == seriesID);
+            var availableSurfaces = _context.SeriesSurface
+                                        .Where(ss => ss.SeriesID == seriesID)?
+                                        .ToArray();
+            var surfaceIndex = 0;
+            var match = new MatchUploadViewModel
+            {
+                StartTime = series.StartDate,
+                SeriesID = seriesID,
+                SurfaceID = availableSurfaces is null ? 1 : availableSurfaces[surfaceIndex].SurfaceID,
+                TeamIDs = teams
+            };
+            //surfaceIndex++; //surfaceIndex can be incremented if we plan to rework ladder match logic (to make more than 1 match)
+
+            var res = await _matchService.PostMatchAsync(match);
+            createdMatches.Add(res);
+            return createdMatches;
+        }
+
+        //2nd Part of Autoscheduling
+
+        private async Task<IEnumerable<MatchViewModel>> ScheduleDoubleElimination(int seriesID)
+        {
+            throw new NotImplementedException();
+        }
+        private async Task<IEnumerable<MatchViewModel>> ScheduleSingleElimination(int seriesID)
+        {
+            var createdMatches = new List<MatchViewModel>();
+            var series = _context.Series.FirstOrDefault(s => s.ID == seriesID);
+            var teams = _context.SeriesTeam
+               .Where(st => st.SeriesID == seriesID)
+               .OrderByDescending(st => st.Win);
+
+            SeriesScheduleViewModel schedule = _context.SeriesSchedule
+                            .FirstOrDefault(ss => ss.ID ==
+                                _context.Series
+                                    .FirstOrDefault(s => s.ID == seriesID)
+                                    .ScheduleID);
```

Comment:
```suggestion
            SeriesScheduleViewModel schedule = _context.SeriesSchedule
                            .FirstOrDefault(ss => ss.ID == ss.Series.ScheduleID);
```
?
Also why does a SeriesSchedule have a ICollection of serieses? Shouldnt it only map to 1 series?

Reply:
Unfortunately no. This was the proposed implementation for saving space a while back by @EjPlatzer. SeriesSchedule, since many activities and series' will share a similar time block i.e. Weekdays from 4:00-9:00 PM it would be more efficient to have a PUT route that simply assigns an existing series schedule to that series. Otherwise, just make a new one. Because of this backwards relation, I also can't do the ss.Series.ScheduleID since there are multiple Series per series schedule as you stated.